### PR TITLE
Replace "0" with random Partition key (#7178)

### PR DESCRIPTION
**Pull request by soo-toance**
before 

var ehClient = EventHubClient.CreateFromConnectionString(connectionString);
var partitionSender = ehClient.CreatePartitionSender("0");

after:

var partitions = await this.GetPartitionsAsync(ehClient);
var partitionId = partitions[new Random().Next(partitions.Length)];
var partitionSender = ehClient.CreatePartitionSender(partitionId);

### CODE REVIEW

**Review by jsquire — APPROVED** [review]
Hi @soo-toance.  Thank you very much for your contribution.  I left one minor suggestion, but the changes look good to me, overall.   If you don't mind addressing the one open comment, I'd be happy to merge this in.

File: sdk/eventhub/Microsoft.Azure.EventHubs/tests/Client/DataBatchTests.cs [hunk 1]
```diff
@@ -79,9 +79,14 @@ await using (var scope = await EventHubScope.CreateAsync(1))
             {
                 var connectionString = TestUtility.BuildEventHubsConnectionString(scope.EventHubName);
                 var ehClient = EventHubClient.CreateFromConnectionString(connectionString);
-                var partitionSender = ehClient.CreatePartitionSender("0");
+                var partitionSender = default(PartitionSender);
+
                 try
                 {
+                    var partitions = await GetPartitionsAsync(ehClient);
+                    var partitionId = partitions[new Random().Next(partitions.Length)];
```
[the remark this answers is not in the document]
Comment (soo-toance) [reply sdk/eventhub/Microsoft.Azure.EventHubs/tests/Client/DataBatchTests.cs:87]: hello : ) thanks for the suggestion! i'll fix it.

[the remark this answers is not in the document]
Comment (soo-toance) [reply sdk/eventhub/Microsoft.Azure.EventHubs/tests/Client/DataBatchTests.cs]: hello @jsquire  i  fix it below. (e612a8bceafabf0e2f91ad97892871703aa5a473)  Thanks!

[the remark this answers is not in the document]
Comment (jsquire) [reply sdk/eventhub/Microsoft.Azure.EventHubs/tests/Client/DataBatchTests.cs]: Thanks, @soo-toance!  The change looks good to me,  I'll go ahead and merge things in.   Thank you, again, for your contribution and helping to improve the Azure SDKs .

**Pull request merged**